hathor/p2p/resources/add_peers.py

```diff
@@ -56,8 +56,8 @@ def already_connected(connection_string: str) -> bool:
             # determines if given connection string is already among connected or connecting peers
             endpoint_url = connection_string.replace('//', '')
 
-            # remove peers that we're already trying to connect
-            if endpoint_url in self.manager.connections.connecting_peers:
+            # ignore peers that we're already trying to connect
+            if endpoint_url in set(self.manager.connections.get_not_ready_connections()):
```

Comment:
I guess it is right because the `endpoint_url` is the key of the `connecting_peers`. But it is better to check whether it's being properly tested.

Reply:
I understand the typing problem. But I thought there was also logic problem of not including "ready" connections, since `connecting_peers` will not include peers that are "ready".

Reply:
Fixed in #292 .